SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -29,52 +29,62 @@ ComponentReader::ComponentReader(USpatialNetDriver* InNetDriver, FObjectReferenc
 
 void ComponentReader::ApplyComponentData(const Worker_ComponentData& ComponentData, UObject* Object, USpatialActorChannel* Channel, bool bIsHandover)
 {
-	Schema_Object* ComponentObject = Schema_GetComponentDataFields(ComponentData.schema_type);
-
-	if (bIsHandover)
+	if (Object->IsPendingKill())
 	{
-		ApplyHandoverSchemaObject(ComponentObject, Object, Channel, true);
-	}
-	else
-	{
-		ApplySchemaObject(ComponentObject, Object, Channel, true);
+		return;
 	}
-}
 
-void ComponentReader::ApplyComponentUpdate(const Worker_ComponentUpdate& ComponentUpdate, UObject* Object, USpatialActorChannel* Channel, bool bIsHandover)
-{
-	Schema_Object* ComponentObject = Schema_GetComponentUpdateFields(ComponentUpdate.schema_type);
+	Schema_Object* ComponentObject = Schema_GetComponentDataFields(ComponentData.schema_type);
 
 	TArray<Schema_FieldId> ClearedIds;
-	ClearedIds.SetNum(Schema_GetComponentUpdateClearedFieldCount(ComponentUpdate.schema_type));
-	Schema_GetComponentUpdateClearedFieldList(ComponentUpdate.schema_type, ClearedIds.GetData());
+	TArray<uint32> UpdatedIds;
+	UpdatedIds.SetNumUninitialized(Schema_GetUniqueFieldIdCount(ComponentObject));
+	Schema_GetUniqueFieldIds(ComponentObject, UpdatedIds.GetData());
```

Comment:
For initial data, we should probably iterate over all the fields regardless of whether they're present (i.e. include empty arrays).
This issue is not introduced by this PR though, so feel free to merge without making that change, but we should add a ticket.

Reply:
UNR-959